django/db/migrations/serializer.py

```diff
@@ -0,0 +1,396 @@
+
+
+class FrozensetSerializer(SequenceSerializer):
+    def _format(self):
+        return "frozenset([%s])"
+
+
+class SetSerializer(SequenceSerializer):
+    def _format(self):
+        # Don't use the literal "{%s}" as it doesn't support empty set
+        return "set([%s])"
+
+
+class TupleSerializer(SequenceSerializer):
+    def _format(self):
+        # When len(value)==0, the empty tuple should be serialized as
+        # "()", not "(,)" because (,) is invalid Python syntax.
+        return "(%s)" if len(self.value) != 1 else "(%s,)"
+
+
+class DictionarySerializer(BaseSerializer):
+    def serialize(self):
+        imports = set()
+        strings = []
+        for k, v in sorted(self.value.items()):
+            k_string, k_imports = serializer_factory(k).serialize()
+            v_string, v_imports = serializer_factory(v).serialize()
+            imports.update(k_imports)
+            imports.update(v_imports)
+            strings.append((k_string, v_string))
+        return "{%s}" % (", ".join("%s: %s" % (k, v) for k, v in strings)), imports
+
+
+class EnumSerializer(BaseSerializer):
+    def serialize(self):
+        enum_class = self.value.__class__
+        module = enum_class.__module__
+        imports = {"import %s" % module}
+        v_string, v_imports = serializer_factory(self.value.value).serialize()
+        imports.update(v_imports)
+        return "%s.%s(%s)" % (module, enum_class.__name__, v_string), imports
+
+
+class DatetimeSerializer(BaseSerializer):
+    def serialize(self):
+        value_repr = self.serialize_datetime()
+        imports = ["import datetime"]
+        if self.value.tzinfo is not None:
+            imports.append("from django.utils.timezone import utc")
+        return value_repr, set(imports)
+
+    def serialize_datetime(self):
+        """
+        Returns a serialized version of a datetime object that is valid,
+        executable python code. It converts timezone-aware values to utc with
+        an 'executable' utc representation of tzinfo.
+        """
+        if self.value.tzinfo is not None and self.value.tzinfo != utc:
+            self.value = self.value.astimezone(utc)
+        value_repr = repr(self.value).replace("<UTC>", "utc")
+        if isinstance(self.value, datetime_safe.datetime):
+            value_repr = "datetime.%s" % value_repr
+        return value_repr
+
+
+class DateSerializer(BaseSerializer):
+    def serialize(self):
+        value_repr = repr(self.value)
+        if isinstance(self.value, datetime_safe.date):
+            value_repr = "datetime.%s" % value_repr
+        return value_repr, {"import datetime"}
+
+
+class TimeSerializer(BaseSerializer):
+    def serialize(self):
+        value_repr = repr(self.value)
+        if isinstance(self.value, datetime_safe.time):
+            value_repr = "datetime.%s" % value_repr
+        return value_repr, {"import datetime"}
+
+
+class TimeDeltaSerializer(BaseSerializer):
```

Comment:
In accordance with the naming of `DatetimeSerializer` this should be `TimedeltaSerializer`.